### ISSUE TITLE
Reapplying tether classes causes CSS animations to restart

### ISSUE THREAD

**Issue by j3tan** [danreeves/react-tether#201]
I'll preface this by saying that it is not a good pattern for consumers of `react-tether` to create CSS selectors on library-controlled classes but it sometimes is necessary to apply "directional-specific" classes to bind "caret" css for tooltips and to use custom "expand/collapse" animations for the different positions.

Example:
```
.tether-element-attached-top .contents {
  animation-name: expand-down;
  animation-duration: 0.5s;
}
.tether-element-attached-bottom .contents {
  animation-name: expand-up;
  animation-duration: 0.5s;
}
```

It would be preferable if `react-tether` did not remove tether classes while doing an update so that CSS animations do not restart when the position of the tether has not changed. Currently, the code removes all tether-classes in `TetherComponent`, see latest code from #189 
```
var classStr = className || '';
if (this._elementParentNode.className !== classStr) {
    this._elementParentNode.className = classStr;
}
```
Then re-applies the same classes in `tether.position()` which causes **CSS animations to fire again.** This behavior was not present (perhaps overlooked) before when a `className` wasn't passed to the `TetherComponent` but I suspect this is done to compute the correct bounding boxes for tether positioning.

Old Code:
```
if (className) {
    this._elementParentNode.className = className;
}
```

Related CodePen (tether@1.0.2): https://codepen.io/j3tan/pen/exGRPw
Related CodePen (tether@2.0.2): https://codepen.io/j3tan/pen/JjPMxQx

Ideally, tether classes should only be removed/modified if they have changed during the update.

**Comment by danreeves** [comment]
Thanks for the detailed issue! This shouldn't be too hard to fix using the classList api

**Comment by danreeves** [comment]
Published in react-tether@2.0.4
